Microsoft.Toolkit.Services/Services/Twitter/TwitterDataProvider.cs

```diff
@@ -341,93 +308,58 @@ public void Logout()
         /// <returns>Success or failure.</returns>
         public async Task<bool> TweetStatusAsync(TwitterStatus status, params IRandomAccessStream[] pictures)
         {
-            try
-            {
-                var mediaIds = string.Empty;
+            var mediaIds = string.Empty;
 
-                if (pictures != null && pictures.Length > 0)
+            if (pictures != null && pictures.Length > 0)
+            {
+                var ids = new List<string>();
+                foreach (var picture in pictures)
                 {
-                    var ids = new List<string>();
-                    foreach (var picture in pictures)
-                    {
-                        ids.Add(await UploadPictureAsync(picture));
-                    }
-
-                    mediaIds = "&media_ids=" + string.Join(",", ids);
+                    ids.Add(await UploadPictureAsync(picture));
                 }
 
-                var uri = new Uri($"{BaseUrl}/statuses/update.json?{status.RequestParameters}{mediaIds}");
-
-                TwitterOAuthRequest request = new TwitterOAuthRequest();
-                await request.ExecutePostAsync(uri, _tokens);
-
-                return true;
+                mediaIds = "&media_ids=" + string.Join(",", ids);
             }
-            catch (WebException wex)
-            {
-                HttpWebResponse response = wex.Response as HttpWebResponse;
-                if (response != null)
-                {
-                    if ((int)response.StatusCode == 429)
-                    {
-                        throw new TooManyRequestsException();
-                    }
 
-                    if (response.StatusCode == HttpStatusCode.Unauthorized)
-                    {
-                        throw new OAuthKeysRevokedException();
-                    }
-                }
+            var uri = new Uri($"{BaseUrl}/statuses/update.json?{status.RequestParameters}{mediaIds}");
 
-                throw;
-            }
-        }
+            TwitterOAuthRequest request = new TwitterOAuthRequest();
+            await request.ExecutePostAsync(uri, Tokens, SigntureManager);
 
+            return true;
+        }
+#else
```

Comment:
This could also exist on UWP in cases where code needs to be shared with other platforms, no?